nucypher/utilities/logging.py

```diff
@@ -163,15 +176,43 @@ def _ensure_dir_exists(path):
     pathlib.Path(path).mkdir(parents=True, exist_ok=True)
 
 
-def get_json_file_observer(name="nucypher.json", path=USER_LOG_DIR):
+def get_json_file_observer(name=DEFAULT_JSON_LOG_FILENAME, path=USER_LOG_DIR):
     _ensure_dir_exists(path)
     logfile = LogFile(name=name, directory=path, rotateLength=MAXIMUM_LOG_SIZE, maxRotatedFiles=MAX_LOG_FILES)
     observer = jsonFileLogObserver(outFile=logfile)
     return observer
 
 
-def get_text_file_observer(name="nucypher.log", path=USER_LOG_DIR):
+def get_text_file_observer(name=DEFAULT_LOG_FILENAME, path=USER_LOG_DIR):
     _ensure_dir_exists(path)
     logfile = LogFile(name=name, directory=path, rotateLength=MAXIMUM_LOG_SIZE, maxRotatedFiles=MAX_LOG_FILES)
     observer = FileLogObserver(formatEvent=formatEventAsClassicLogText, outFile=logfile)
     return observer
+
+
+class Logger(TwistedLogger):
+    """Drop-in replacement of Twisted's Logger, patching the emit() method to tolerate inputs with curly braces,
+    i.e., not compliant with PEP 3101.
+
+    See Issue #724 and, particularly, https://github.com/nucypher/nucypher/issues/724#issuecomment-600190455"""
+
+    CURLY_BRACES_REGEX = re.compile('{+|}+')
+
+    @classmethod
+    def escape_format_string(cls, string):
+        """
+        Escapes curly braces from a PEP-3101's format string when there's a sequence of odd length
+        """
+
+        def escape_group_of_curly_braces(match):
+            curlies = match.group()
+            if len(curlies) % 2 == 1:
+                curlies += curlies
+            return curlies
+
+        escaped_string = cls.CURLY_BRACES_REGEX.sub(escape_group_of_curly_braces, string)
+        return escaped_string
+
+    def emit(self, level, format=None, **kwargs):
```

Comment:
So does this essentially mean that we won't ever use the formatting functionality of `twisted`'s `Logger`? All the curly braces will be escaped?

Reply:
Yes, pretty much. If someone is interested in that they can always use the original Logger. The problem is that sometimes we directly log messages coming from 3rd party libraries, like exceptions or responses, which may not be adequate for this use, and you never know in advance.